### PR TITLE
Lua: added PDmod binding for VTOL tuning

### PR DESCRIPTION
**Pull request by tridge**
This allows for a lua script to do initial VTOL tuning for copters and quadplanes

### CODE REVIEW

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 1]
```diff
@@ -504,3 +504,7 @@ singleton AP_Follow method get_target_location_and_velocity boolean Location'Nul
 singleton AP_Follow method get_target_location_and_velocity_ofs boolean Location'Null Vector3f'Null
 singleton AP_Follow method get_target_heading_deg boolean float'Null
 
+include AC_AttitudeControl/AC_AttitudeControl.h depends APM_BUILD_TYPE(APM_BUILD_ArduPlane)||APM_BUILD_COPTER_OR_HELI
+singleton AC_AttitudeControl depends APM_BUILD_TYPE(APM_BUILD_ArduPlane)||APM_BUILD_COPTER_OR_HELI
+singleton AC_AttitudeControl method get_rpy_PDmod boolean Vector3f'Null
```
Comment (IamPete1) [inline libraries/AP_Scripting/generator/description/bindings.desc:509]: No need to return always true if you do this:
```suggestion
singleton AC_AttitudeControl method get_rpy_PDmod void Vector3f'Ref
```

**Review by IamPete1 — CHANGES_REQUESTED** [review]
Needs docs

**Comment by IamPete1** [comment]
We will only get a value is the slew rate limit set, maybe the first call to this new method could enable slew rate calculation. Or I guess your script could just set a very large slew limit if not setup.

File: libraries/AP_Scripting/generator/description/bindings.desc (hunk 1 above)
Comment (tridge) [reply libraries/AP_Scripting/generator/description/bindings.desc:509]: thanks, fixed

**Comment by tridge** [comment]
> Or I guess your script could just set a very large slew limit if not setup.

yes, the script has a QUIK_OSC_SMAX parameter and it sets the SMAX to that, and reverts when finished

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp [hunk 2]
```diff
@@ -1129,3 +1131,15 @@ bool AC_AttitudeControl::pre_arm_checks(const char *param_prefix,
...
+/*
+  get the PDmod value for roll, pitch and yaw, for oscillation
+   detection in lua scripts
+*/
+bool AC_AttitudeControl::get_rpy_PDmod(Vector3f &PDmod)
+{
+    PDmod = Vector3f(get_rate_roll_pid().get_pid_info().Dmod,
+                     get_rate_pitch_pid().get_pid_info().Dmod,
+                     get_rate_yaw_pid().get_pid_info().Dmod);
+    return true;
```
Comment (IamPete1) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:1144]: No longer using this return value.

Comment (tridge) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.cpp]: fixed

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp [hunk 3]
```diff
@@ -1129,3 +1131,14 @@ bool AC_AttitudeControl::pre_arm_checks(const char *param_prefix,
...
     return true;
 }
+
+/*
+  get the PDmod value for roll, pitch and yaw, for oscillation
+   detection in lua scripts
+*/
+void AC_AttitudeControl::get_rpy_PDmod(Vector3f &PDmod)
+{
+    PDmod = Vector3f(get_rate_roll_pid().get_pid_info().Dmod,
```
Comment (tridge) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:1141]: change to 3 refs

**Pull request merged**